tests/epyccel/test_epyccel_lists.py

```diff
@@ -222,3 +222,53 @@ def clear_3():
     pyccel_result = epyc_clear_3()
     python_result = clear_3()
     assert python_result == pyccel_result
+
+def test_extend_basic(language):
+    def f():
+        a = [1, 2, 3]
+        b = [4, 5, 6]
+        a.extend(b)
+        return a
+
+    epyc_f = epyccel(f, language=language)
+    assert f() == epyc_f()
+
+def test_extend_multiple(language):
+    def f():
+        a = [1, 2, 3]
+        a.extend([4, 5])
+        a.extend([6, 7, 8, 9])
+        a.extend([10])
+        return a
+
+    epyc_f = epyccel(f, language=language)
+    assert f() == epyc_f()
+
+def test_extend_nested_list(language):
+    def f():
+        a = [[1, 2, 3]]
+        a.extend([[4, 5, 6]])
+        return a
+
+    epyc_f = epyccel(f, language=language)
+    assert f() == epyc_f()
+
+def test_extend_tuple_with_list(language):
+    def f():
+        a = [1, 2, 3]
+        b = (4, 5, 6)
+        a.extend(b)
+        return a
```

Comment:
Nice that you are testing an homogeneous tuple as an argument! Can you try passing some other iterable? We have ranges and ndarrays.

Reply:
tests updated in c4bffd9